.github/workflows/ci.yml

```
@@ -0,0 +1,34 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main   
+  pull_request:  
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+          components: clippy
+
+      - name: Install cargo-hack
+        run: cargo install cargo-hack
+
+      - name: Check compilation
+        run: cargo hack check --all-features --all-targets
+
+      - name: Check formatting
+        run: cargo +nightly fmt --all --check
+
+      - name: Run Clippy
+        run: cargo hack clippy --all-features --all-targets
```

We need not run `cargo hack` here, just `cargo clippy` should suffice:

```suggestion
      - name: Run Clippy
        run: cargo clippy --all-features --all-targets
```